### ISSUE TITLE
slow loadtimes after Firefox-fix

### ISSUE THREAD

**Issue by Benman2785** [JohnImril/diablo_web_simple#5]
Hi,

on my slow work PC the loadtimes are now very high with diablo_web_simple

in the video i compare on same machine diablo_web_simple (first) and diablo_web (second)
https://youtu.be/9hs7DDZdrE0

you can see - it takes like 1min loading the menu and 45s to load a multiplayer game in diablo_web_simple -> while diablo_web is way faster here

even on my home pc (Ryzen 7 5800X3D) it takes 3-5s to load a MP game instead of 2s

**Comment by Benman2785** [comment]
https://youtu.be/xpbHxlJuaqs

here on my home PC -> initialising takes 3s instead of under 1s // loading a SP or MP game (the first time in that session) takes 3s instead of under 1s

**Comment by Benman2785** [comment]
tested it -> with that fix we have no savegames again in Firefox; so i think you should add 

> 	if (spawn && !mpq) {
> 		await load_diabdat(api, fs);
> 	}

again ;)

**Comment by Benman2785** [comment]
when i first start the game it downloads diabdat.mpq -> but while playing (like re-watching credits, starting a sp/mp game, talking to NPCs it always downloads a chunk of diabdat.mpq (between 1-7mb) -> this is why the loadtimes are bigger.

so instead of using the fully downloaded mpq from cache it downloads chunks

will test now if Chrome behaves the same way too

**Comment by Benman2785** [comment]
maybe it is related to this:

While offline, a simple change in the URL could be a problem

Eg: http://localhost:8080/app doesn't work on Firefox/IE
but http://localhost:8080/app/ works in Firefox/IE

**Comment by Benman2785** [comment]
now I am 100% sure that slow load is a Firefox problem. I will try to configure nginx correctly and test again.

but you should add you load-fix again:
if (spawn && !mpq) {
await load_diabdat(api, fs);
}

**Comment by Benman2785** [comment]
https://support.mozilla.org/en-US/questions/1475203

maybe we get an answer there

**Comment by Benman2785** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1932581

**Comment by JohnImril** [comment]
added a temporary solution if (!navigator.userAgent.includes("Firefox")) await load_diabdat(api, fs); just so it works for now, I continue to sort out the problem

**Comment by Benman2785** [comment]
btw - with your temporary solution you need to have diabdat.mpq in ./diablo_web_simple (for Chrome) and ./diablo_web_simple/assets (for Firefox)

so currently i still run your code:
if (spawn && !mpq) {
await load_diabdat(api, fs);
}

**Comment by Benman2785** [comment]
oh - i will wait for you to implement more features to test or until Mozilla fixes Firefox
so if i dont write everyday i still have not lost the interest :)

**Comment by JohnImril** [comment]
Hi, I noticed through testing that splitting a file into chunks no longer causes an error, saving itself still doesn't work, but at least I understood in which direction to move. I hope there will be a solution soon

**Comment by Benman2785** [comment]
and in bugzilla they said that Firefox has a problem with cacheing chunks of files that are not "in order" ->

one way is to leave it as it is and wait for Firefox to fix your current problem with cache (as all other browsers work fine)
or to rewrite the code to download diabdat.mpq into cache at first "play"

**Comment by JohnImril** [comment]
Hi, please check again, you need to clear the app cache and data first again.

**Comment by Benman2785** [comment]
Hi, and happy first advent :)

tested and its worse now as Chrome too behaves like Firefox now -> regardless of browser it always loads the chunks now instead of using cache :/

i think you should revert to "Commit 83edec7" & update package.json && than we wait for Firefox to rewrite their cache-system as stated in my bugzilla post. But nice to see you added my "stretched Fullscreen Mode" :)

Sry that i made you so much work

**Comment by JohnImril** [comment]
yes, both browsers download the file and don't cache it, I understand, but do they give errors and do the saves work?

**Comment by JohnImril** [comment]
fixed saves

**Comment by Benman2785** [comment]
sry - i did watch Christmas movies at the evening.

I will test this in 6h

**Comment by Benman2785** [comment]
tested saves in previous and current commit = both work fine.
also: cache now works fine in Firefox
still works great in Chrome :)

thank you

**Comment by JohnImril** [comment]
Wow, yeah, then I should close this issue?

**Comment by JohnImril** [comment]
dependencies also updated

**Comment by Benman2785** [comment]
yeah - it works now :)
